src/main/java/com/epam/training/designpatterns/fastfoodrestaurant/food/Condiment.java

```diff
@@ -0,0 +1,22 @@
+package com.epam.training.designpatterns.fastfoodrestaurant.food;
+
+public abstract class Condiment {
```

Comment:
Helyesen használt decorator minta. Tankönyv szerint jogos az ősosztály használata, de az életben, nagyobb projektben lehet, hogy nem csinálnék ősosztályt az extráknak, mert túl erős csatolást vezet be, refactoringnak nem annyira barátja ilyenkor a kód. Tudom, ezzel megszegjük a DRY elvet...erről még lesz szó máshol.

Reply:
Valóban, azt akartam inkább elkerülni ezzel, hogy az OrderBuilder withCondiments metódusába ne lehessen beadni pl egy ételt mint a hotdog vagy chips. Ha Food-ot várok Condiment helyett, akkor compile-time hiba nem dobódna azzal, hogy egy Chips-et adnak be, hanem később szállna el a program amikor valami ehhez hasonló hívódna meg: new Chips(hotdog)